am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java

```diff
@@ -1,42 +1,50 @@
 package uk.gov.hmcts.reform.amlib.helpers;
 
+import com.fasterxml.jackson.core.JsonPointer;
 import uk.gov.hmcts.reform.amlib.enums.Permission;
+import uk.gov.hmcts.reform.amlib.models.ExplicitAccessGrant;
 import uk.gov.hmcts.reform.amlib.models.ExplicitAccessMetadata;
-import uk.gov.hmcts.reform.amlib.models.ExplicitAccessRecord;
 
+import java.util.Map;
 import java.util.Set;
 
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ACCESSOR_ID;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ACCESS_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SECURITY_CLASSIFICATION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
+
 public final class TestDataFactory {
 
     private TestDataFactory() {
         //NO-OP
     }
 
-    public static ExplicitAccessRecord createRecord(String resourceId,
-                                                    String accessorId,
-                                                    Set<Permission> explicitPermissions) {
-        return ExplicitAccessRecord.builder()
+    public static ExplicitAccessGrant createRecord(String resourceId,
+                                                   String accessorId,
+                                                   Map<JsonPointer, Set<Permission>> attributePermissions) {
+        return ExplicitAccessGrant.builder()
             .resourceId(resourceId)
             .accessorId(accessorId)
-            .explicitPermissions(explicitPermissions)
-            .accessType(TestConstants.ACCESS_TYPE)
-            .serviceName(TestConstants.SERVICE_NAME)
-            .resourceType(TestConstants.RESOURCE_TYPE)
-            .resourceName(TestConstants.RESOURCE_NAME)
-            .attribute("")
-            .securityClassification(TestConstants.SECURITY_CLASSIFICATION)
+            .accessType(ACCESS_TYPE)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attributePermissions(attributePermissions)
+            .securityClassification(SECURITY_CLASSIFICATION)
             .build();
     }
 
     public static ExplicitAccessMetadata createMetadata(String resourceId) {
         return ExplicitAccessMetadata.builder()
             .resourceId(resourceId)
-            .accessorId(TestConstants.ACCESSOR_ID)
-            .accessType(TestConstants.ACCESS_TYPE)
-            .serviceName(TestConstants.SERVICE_NAME)
-            .resourceType(TestConstants.RESOURCE_TYPE)
-            .resourceName(TestConstants.RESOURCE_NAME)
-            .attribute("")
+            .accessorId(ACCESSOR_ID)
+            .accessType(ACCESS_TYPE)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attribute("/")
```

Comment:
I guess we can keep `""`. Sadly `/` is a bit wired construct in JSON Pointer spec. It points to `""` attribute that is almost never used.